Content.Shared/Construction/SharedFlatpackSystem.cs

```diff
@@ -136,6 +136,27 @@ public void SetupFlatpack(Entity<FlatpackComponent?> ent, Entity<MachineBoardCom
         Appearance.SetData(ent, FlatpackVisuals.Machine, MetaData(machineBoard).EntityPrototype?.ID ?? string.Empty);
     }
 
+    public void SetupFlatpack(Entity<FlatpackComponent?> ent, Entity<ComputerBoardComponent?> machineBoard)
+    {
+        if (!Resolve(ent, ref ent.Comp) || !Resolve(machineBoard, ref machineBoard.Comp))
+            return;
+
+        if (machineBoard.Comp.Prototype is not { } machinePrototypeId)
+            return;
+
+        var comp = ent.Comp!;
+        var machinePrototype = PrototypeManager.Index(machinePrototypeId);
+
+        var meta = MetaData(ent);
+        _metaData.SetEntityName(ent, Loc.GetString("flatpack-entity-name", ("name", machinePrototype.Name)), meta);
+        _metaData.SetEntityDescription(ent, Loc.GetString("flatpack-entity-description", ("name", machinePrototype.Name)), meta);
+
+        comp.Entity = machinePrototypeId;
+        Dirty(ent, comp);
+
+        Appearance.SetData(ent, FlatpackVisuals.Machine, MetaData(machineBoard).EntityPrototype?.ID ?? string.Empty);
+    }
```

Comment:
this method is a 1:1 copy of the one for machine boards. You can 100% simplify this by doing TryComps with optional nullable arguments for components. The only difference between the two is how you get the prototype for the machine/computer itself.